apps/dashboard/src/components/primitives/control-input/hooks/use-variables.ts

```diff
@@ -35,15 +46,13 @@ export function useVariables(viewRef: React.RefObject<EditorView>, onChange: (va
         isUpdatingRef.current = true;
         const { from, to } = selectedVariable;
         const view = viewRef.current;
+        const strippedValue = stripBrackets(newValue);
+        let newVariableText = newValue.match(/^\{+.*\}+$/) ? normalizeVariableSyntax(newValue) : `{{${newValue}}}`;
 
-        // Function to normalize variable syntax by reducing multiple brackets to two
-        const normalizeVariableSyntax = (value: string): string => {
-          const strippedValue = value.replace(/[{}]/g, '').trim();
-
-          return `{{${strippedValue}}}`;
-        };
-
-        const newVariableText = newValue.match(/^\{+.*\}+$/) ? normalizeVariableSyntax(newValue) : `{{${newValue}}}`;
+        if (!strippedValue) {
+          // if the value is empty, remove the variable
+          newVariableText = '';
+        }
```

Comment:
this hook is used in the `control-input` and it handles the variable update; here we say that when the value is "empty" we want to remove it from the editor